lib/bolt/executor.rb

```diff
@@ -1,16 +1,26 @@
+require 'set'
+require 'logger'
 require 'concurrent'
 require 'bolt/result'
 require 'bolt/config'
+require 'bolt/formatter'
 
 module Bolt
   class Executor
     def initialize(config = Bolt::Config.new)
       @config = config
+      @logger = Logger.new(config[:log_destination])
+      @logger.progname = 'executor'
+      @logger.level = config[:log_level]
+      @logger.formatter = Bolt::Formatter.new
+      @initialized = Set.new
     end
 
     def from_uris(nodes)
       nodes.map do |node|
-        Bolt::Node.from_uri(node, config: @config)
+        result = Bolt::Node.from_uri(node, config: @config)
+        initialize_transport(result.class)
```

Comment:
Move this to `execute` so that `from_uris` doesn't have side-effects. Also makes more sense if we wanted to add `dispose_transport` lifecycle method in the future.

Reply:
Done